src/apps/Altinn.AccessManagement/src/Altinn.AccessMgmt.Core/HostedServices/Services/PartySyncService.cs

```diff
@@ -73,34 +67,23 @@ public async Task SyncParty(ILease lease, CancellationToken cancellationToken)
 
             foreach (var item in page?.Content.Data ?? [])
             {
-                try
+                var data = item switch
                 {
-                    if (!_supportedEntityTypes.Any(e => e.Equals(item.PartyType, StringComparison.InvariantCultureIgnoreCase)))
-                    {
-                        continue;
-                    }
-
-                    var entity = ConvertPartyModel(item, cancellationToken: cancellationToken);
-                    if (entity is { })
-                    {
-                        if (!seen.Add(entity.Id) || seen.Count > _bulkSize)
-                        {
-                            await Flush();
-                        }
-
-                        bulk.Add(entity);
-                        bulkLookup.AddRange(ConvertPartyModelToLookup(item));
-                    }
-                    else
-                    {
-                        _logger.LogWarning("skipped adding entity of type '{type}' with id '{id}'", item.PartyType, item.PartyUuid);
-                    }
-                }
-                catch (Exception ex)
+                    Person person => MapPerson(person),
+                    Organization organization => MapOrganization(organization),
+                    SelfIdentifiedUser selfIdentifiedUser => MapSelfIdentifiedUser(selfIdentifiedUser),
+                    SystemUser systemUser => MapSystemUser(systemUser),
+                    EnterpriseUser enterpriseUser => MapEnterpriseUser(enterpriseUser),
+                    _ => throw new InvalidDataException($"Unkown Party type {item.Type}"),
```

Comment:
Throw her vil da blokkere all import gitt at Register publiserer data for en ny type?

Reply:
Er vel fortsatt usikkert hvordan de _nye_ SI brukerene fra ID-Porten vil komme inn. Jeg antar de får sin egen type.
De vil hvertfall potensielt ha eget sett med metadata og en ny type ekstern identifikator.

 få på plass import pipeline for en gitt type(r) hadde vært nice ja.